cmd/kubeadm/app/cmd/init.go

```diff
@@ -163,6 +163,17 @@ func newCmdInit(out io.Writer, initOptions *initOptions) *cobra.Command {
 		if len(initRunner.Options.SkipPhases) == 0 {
 			initRunner.Options.SkipPhases = data.cfg.SkipPhases
 		}
+		// dump AddOns status by skip-phases
+		if isStringInSlice("addon/coredns", initRunner.Options.SkipPhases) {
+			data.cfg.AddOns["coredns"] = false
+		}
+		if isStringInSlice("addon/kube-proxy", initRunner.Options.SkipPhases) {
+			data.cfg.AddOns["kube-proxy"] = false
+		}
+		if isStringInSlice("addon", initRunner.Options.SkipPhases) {
+			data.cfg.AddOns["coredns"] = false
+			data.cfg.AddOns["kube-proxy"] = false
+		}
```

Comment:
if the user deploys all addons by just calling `kubeadm init` without skipping phases, does it mean we have to set this Addons map key/value to `"someaddonname": true`?

Reply:
> if the user deploys all addons by just calling kubeadm init without skipping phases, does it mean we have to set this Addons map key/value to "someaddonname": true

By default, without the set in the config file, the addon will be pulled anyway, current logic is the skip will only happen if it's set to false explicitly in the config file. @ruquanzhao is that right?

Reply:
Thanks @chendave for explanation. 
In this implement, `skip-phases` will set `addons` false, and the `addons` only take effect when set to false(by `skip-phases` or user set in configfile)

It does seem more reasonable to use some field like `SkippedAddons`.

Reply:
+1 for `SkippedAddons`